src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs

```diff
@@ -303,20 +303,46 @@ private async Task DownloadContentAndUpdateCache(GithubEntry parentGithubEntry,
                 .Where(x => !string.IsNullOrWhiteSpace(x.Download_url) 
                 && !string.IsNullOrWhiteSpace(x.Name.Split(new char[] { '.' }).LastOrDefault()));
 
-            if (selectedGithubFiles.Any())
+            var assemblyName = Guid.NewGuid().ToString();
+            var expectedFiles = new string[] { "csx", "package.json", "metadata.json" , "kustoClusterMappings"};
+
+            // Identify files to be downloaded
+            if (!selectedGithubFiles.Any(x => expectedFiles.Any(y => x.Name.Contains(y, StringComparison.CurrentCultureIgnoreCase))))
+            {
+                return;
+            }
```

Comment:
Is this necessary anymore? I had this logic in `GitHubScriptWorkerBase` class so that it would only download csx files and dlls but not the kusto mappings file but now that the downloading is singly handled in `GithubWatcher` class it should be okay to remove this. I think the only logic necessary is to make sure it doesn't download search files which is what it is doing in the first couple lines of this method.

Reply:
Unless your motive is to only download these set files

Reply:
I still want this logic to be here just to make sure we are downloading the files we need...